test/02_gaugeProxyV2_periodizationTest.js

```diff
@@ -0,0 +1,241 @@
+    await expect(GaugeProxyV2.distribute(0, 2)).to.be.revertedWith(
+      "GaugeProxyV2: all period distributions complete"
+    );
+  });
+
+  it("Successfully Distribute(initial) PICKLE to gauges after advancing 7 days", async () => {
+    await advanceSevenDays();
+    await GaugeProxyV2.distribute(0, 2);
+    const pickleGaugeAddr = await GaugeProxyV2.getGauge(pickleLP);
+    const yvecrvGaugeAddr = await GaugeProxyV2.getGauge(pyveCRVETH);
+
+    const pickle = await ethers.getContractAt(
+      "src/yield-farming/pickle-token.sol:PickleToken",
+      pickleAddr
+    );
+
+    let pickleRewards = Number(await pickle.balanceOf(pickleGaugeAddr));
+    console.log("Rewards to Pickle gauge => ", pickleRewards.toString());
+
+    let yvecrvRewards = Number(await pickle.balanceOf(yvecrvGaugeAddr));
+    console.log("Rewards to pyveCRV gauge => ", yvecrvRewards.toString());
+    expect(pickleRewards).to.greaterThan(0);
+    expect(yvecrvRewards).to.greaterThan(0);
+  });
+
+  it("Distribute should fail as all distributions are complete ", async () => {
+    await expect(GaugeProxyV2.distribute(0, 2)).to.be.revertedWith(
+      "GaugeProxyV2: all period distributions complete"
+    );
+  });
+
+  it("Should distribute similar rewards for the periods users have not voted for (advancing 3 weeks)", async () => {
+    await advanceSevenDays();
+    await advanceSevenDays();
+    await advanceSevenDays();
+    console.log("--Advanced 3 weeks--");
+
+    const cid = await GaugeProxyV2.getCurrentPeriodId();
+    const did = await GaugeProxyV2.distributionId();
+
+    console.log("Current Id =>", Number(cid));
+    console.log("Distribution Id =>", Number(did));
+
+    for (let i = did; i < cid; i++) {
+      console.log("Distributing for id => ", i);
+      await GaugeProxyV2.distribute(0, 2);
+      const pickleGaugeAddr = await GaugeProxyV2.getGauge(pickleLP);
+      const yvecrvGaugeAddr = await GaugeProxyV2.getGauge(pyveCRVETH);
+      const pickle = await ethers.getContractAt(
+        "src/yield-farming/pickle-token.sol:PickleToken",
+        pickleAddr
+      );
+
+      const pickleRewards = await pickle.balanceOf(pickleGaugeAddr);
+      console.log("Rewards to Pickle gauge", pickleRewards.toString());
+
+      const yvecrvRewards = await pickle.balanceOf(yvecrvGaugeAddr);
+      console.log("Rewards to pyveCRV gauge", yvecrvRewards.toString());
+    }
+  });
+
+  it("Distribute should fail as all distributions are complete ", async () => {
+    await expect(GaugeProxyV2.distribute(0, 2)).to.be.revertedWith(
+      "GaugeProxyV2: all period distributions complete"
+    );
+  });
+
+  it("Should vote successfully (second voting)", async () => {
+    console.log("-- Voting on LP Gauge with 100% weight --");
+    const gaugeProxyFromUser = GaugeProxyV2.connect(userAddr);
+    populatedTx = await gaugeProxyFromUser.populateTransaction.vote(
```

Comment:
Would it perhaps be more useful to have another user vote (with different weights) and test the feature that votes from past periods persist if untouched?